building for macOS-arm64 but attempting to link with file built for macOS-x86_64

## Steps to Reproduce

1. Update your Mac to Big Sur
2. Run `flutter create bug`
3. Run `flutter run --release` and select `macos` as target


**Actual results:** <!-- what did you see? -->

```
flutter run --release
Multiple devices found:
Pixel XL (mobile)         • HT6A80200946                             • android-arm64  • Android 10 (API 29)
iPhone von Ralph (mobile) • 3fbe4929eec22152a0ffdbaa8c5afef34ebcaafc • ios            • iOS 14.0.1
macOS (desktop)           • macos                                    • darwin-x64     • macOS 11.0.1 20B29 darwin-x64
Web Server (web)          • web-server                               • web-javascript • Flutter Tools
Chrome (web)              • chrome                                   • web-javascript • Google Chrome 86.0.4240.193
[0]: Pixel XL (HT6A80200946)
[1]: iPhone von Ralph (3fbe4929eec22152a0ffdbaa8c5afef34ebcaafc)
[2]: macOS (macos)
[3]: Web Server (web-server)
[4]: Chrome (chrome)
Please choose one (To quit, press "q/Q"): 2
Running "flutter pub get" in foo...                                278ms
Launching lib/main.dart on macOS in release mode...
Command CompileSwift failed with a nonzero exit code
ld: warning: ignoring file /Users/ralphbergmann/development/projects/flutter/foo/macos/Flutter/ephemeral/App.framework/App, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
ld: warning: ignoring file /Users/ralphbergmann/development/projects/flutter/foo/macos/Flutter/ephemeral/FlutterMacOS.framework/FlutterMacOS, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
Undefined symbols for architecture arm64:
  "_OBJC_CLASS_$_FlutterAppDelegate", referenced from:
      type metadata for foo.AppDelegate in AppDelegate.o
  "_OBJC_CLASS_$_FlutterViewController", referenced from:
      objc-class-ref in MainFlutterWindow.o
  "_OBJC_METACLASS_$_FlutterAppDelegate", referenced from:
      _OBJC_METACLASS_$__TtC3foo11AppDelegate in AppDelegate.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
note: Using new build system
note: Building targets in parallel
note: Planning build
note: Constructing build description
** BUILD FAILED **

Building macOS application...
Exception: Build process failed
```

<!-- Finally, paste the output of running `flutter doctor -v` here. -->

```
flutter doctor -v
[✓] Flutter (Channel master, 1.24.0-8.0.pre.252, on macOS 11.0.1 20B29 darwin-x64, locale de-DE)
    • Flutter version 1.24.0-8.0.pre.252 at /Users/ralphbergmann/development/sdks/flutter
    • Framework revision bf2c9dfc05 (10 hours ago), 2020-11-14 22:38:02 -0500
    • Engine revision 3e77d854a7
    • Dart version 2.12.0 (build 2.12.0-45.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    • Android SDK at /Users/ralphbergmann/development/sdks/android-sdk
    • Platform android-30, build-tools 30.0.1
    • ANDROID_HOME = /Users/ralphbergmann/development/sdks/android-sdk
    • Java binary at: /Users/ralphbergmann/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.6953283/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.2, Build version 12B45b
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/ralphbergmann/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.6953283/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Ultimate Edition (version 2020.2.3)
    • IntelliJ at /Users/ralphbergmann/Applications/JetBrains Toolbox/IntelliJ IDEA Ultimate.app
    • Flutter plugin installed
    • Dart plugin version 202.7319.5

[✓] VS Code (version 1.46.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.12.0

[✓] Connected device (4 available)
    • Pixel XL (mobile) • HT6A80200946 • android-arm64  • Android 10 (API 29)
    • macOS (desktop)   • macos        • darwin-x64     • macOS 11.0.1 20B29 darwin-x64
    • Web Server (web)  • web-server   • web-javascript • Flutter Tools
    • Chrome (web)      • chrome       • web-javascript • Google Chrome 86.0.4240.193

• No issues found!
```

<details> 
<summary>logs</summary> 

```bash
[ +118 ms] executing: [/Users/tahatesser/Code/flutter_master/] git -c log.showSignature=false log -n 1
--pretty=format:%H
[  +50 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] bf2c9dfc05a6d281e7d303689f1af9add4030230
[   +1 ms] executing: [/Users/tahatesser/Code/flutter_master/] git tag --points-at HEAD
[  +33 ms] Exit code 0 from: git tag --points-at HEAD
[   +2 ms] executing: [/Users/tahatesser/Code/flutter_master/] git describe --match *.*.* --long --tags
[  +82 ms] Exit code 0 from: git describe --match *.*.* --long --tags
[   +1 ms] 1.24.0-7.0.pre-252-gbf2c9dfc05
[  +25 ms] executing: [/Users/tahatesser/Code/flutter_master/] git rev-parse --abbrev-ref --symbolic @{u}
[  +28 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[   +1 ms] origin/master
[        ] executing: [/Users/tahatesser/Code/flutter_master/] git ls-remote --get-url origin
[  +31 ms] Exit code 0 from: git ls-remote --get-url origin
[   +1 ms] https://github.com/flutter/flutter.git
[ +105 ms] executing: [/Users/tahatesser/Code/flutter_master/] git rev-parse --abbrev-ref HEAD
[  +19 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[   +1 ms] master
[  +16 ms] executing: sw_vers -productName
[  +30 ms] Exit code 0 from: sw_vers -productName
[        ] macOS
[        ] executing: sw_vers -productVersion
[  +28 ms] Exit code 0 from: sw_vers -productVersion
[        ] 11.0.1
[        ] executing: sw_vers -buildVersion
[  +22 ms] Exit code 0 from: sw_vers -buildVersion
[        ] 20B29
[   +1 ms] executing: sysctl hw.optional.arm64
[  +10 ms] Exit code 1 from: sysctl hw.optional.arm64
[        ] sysctl: unknown oid 'hw.optional.arm64'
[  +70 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +8 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +78 ms] executing: /Users/tahatesser/Code/sdk/platform-tools/adb devices -l
[   +9 ms] executing: /usr/bin/xcode-select --print-path
[  +10 ms] Exit code 0 from: /usr/bin/xcode-select --print-path
[        ] /Volumes/Extreme/Xcode.app/Contents/Developer
[   +1 ms] executing: sysctl hw.optional.arm64
[   +5 ms] Exit code 1 from: sysctl hw.optional.arm64
[        ] sysctl: unknown oid 'hw.optional.arm64'
[        ] executing: xcrun xcodebuild -version
[ +581 ms] Exit code 0 from: xcrun xcodebuild -version
[   +1 ms] Xcode 12.2
           Build version 12B45b
[   +3 ms] executing: xcrun xcdevice list --timeout 2
[   +7 ms] xcrun simctl list --json devices
[        ] executing: xcrun simctl list --json devices
[  +16 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +1 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +78 ms] executing: /Users/tahatesser/Code/sdk/platform-tools/adb devices -l
[        ] executing: xcrun xcdevice list --timeout 2
[        ] xcrun simctl list --json devices
[        ] executing: xcrun simctl list --json devices
[  +22 ms] Skipping pub get: version match.
[ +154 ms] Generating
/Users/tahatesser/AndroidStudioProjects/master_flutter/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegis
trant.java
[ +111 ms] Launching lib/main.dart on macOS in release mode...
[  +58 ms] executing: [/Users/tahatesser/AndroidStudioProjects/master_flutter/macos/] xcrun xcodebuild -list -project
Runner.xcodeproj
[  +79 ms] List of devices attached
           EUYTFEUSQSRGDA6D       device usb:336789504X product:RMX2001 model:RMX2001 device:RMX2001L1 transport_id:4
[   +7 ms] List of devices attached
           EUYTFEUSQSRGDA6D       device usb:336789504X product:RMX2001 model:RMX2001 device:RMX2001L1 transport_id:4
[ +229 ms] {
                      "devices" : {
                        "com.apple.CoreSimulator.SimRuntime.iOS-13-7" : [
                          {
                            "dataPath" :
                            "\/Users\/tahatesser\/Library\/Developer\/CoreSimulator\/Devices\/F0FD09FB-9ABB-402D-9712-7A
                            C7278ED742\/data",
                            "logPath" :
                            "\/Users\/tahatesser\/Library\/Logs\/CoreSimulator\/F0FD09FB-9ABB-402D-9712-7AC7278ED742",
                            "udid" : "F0FD09FB-9ABB-402D-9712-7AC7278ED742",
                            "isAvailable" : true,
                            "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-SE--2nd-generation-",
                            "state" : "Shutdown",
                            "name" : "iPhone SE (2nd generation)"
                          }
                        ],
                        "com.apple.CoreSimulator.SimRuntime.watchOS-7-1" : [

                        ],
                        "com.apple.CoreSimulator.SimRuntime.iOS-14-2" : [
                          {
                            "dataPath" :
                            "\/Users\/tahatesser\/Library\/Developer\/CoreSimulator\/Devices\/B46BCEDD-9601-4CD0-8D0B-CE
                            4EF96ACD93\/data",
                            "logPath" :
                            "\/Users\/tahatesser\/Library\/Logs\/CoreSimulator\/B46BCEDD-9601-4CD0-8D0B-CE4EF96ACD93",
                            "udid" : "B46BCEDD-9601-4CD0-8D0B-CE4EF96ACD93",
                            "isAvailable" : true,
                            "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-12",
                            "state" : "Shutdown",
                            "name" : "iPhone 12"
                          }
                        ],
                        "com.apple.CoreSimulator.SimRuntime.tvOS-14-2" : [

                        ]
                      }
                    }
[+1153 ms] {
                      "devices" : {
                        "com.apple.CoreSimulator.SimRuntime.iOS-13-7" : [
                          {
                            "dataPath" :
                            "\/Users\/tahatesser\/Library\/Developer\/CoreSimulator\/Devices\/F0FD09FB-9ABB-402D-9712-7A
                            C7278ED742\/data",
                            "logPath" :
                            "\/Users\/tahatesser\/Library\/Logs\/CoreSimulator\/F0FD09FB-9ABB-402D-9712-7AC7278ED742",
                            "udid" : "F0FD09FB-9ABB-402D-9712-7AC7278ED742",
                            "isAvailable" : true,
                            "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-SE--2nd-generation-",
                            "state" : "Shutdown",
                            "name" : "iPhone SE (2nd generation)"
                          }
                        ],
                        "com.apple.CoreSimulator.SimRuntime.watchOS-7-1" : [

                        ],
                        "com.apple.CoreSimulator.SimRuntime.iOS-14-2" : [
                          {
                            "dataPath" :
                            "\/Users\/tahatesser\/Library\/Developer\/CoreSimulator\/Devices\/B46BCEDD-9601-4CD0-8D0B-CE
                            4EF96ACD93\/data",
                            "logPath" :
                            "\/Users\/tahatesser\/Library\/Logs\/CoreSimulator\/B46BCEDD-9601-4CD0-8D0B-CE4EF96ACD93",
                            "udid" : "B46BCEDD-9601-4CD0-8D0B-CE4EF96ACD93",
                            "isAvailable" : true,
                            "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-12",
                            "state" : "Shutdown",
                            "name" : "iPhone 12"
                          }
                        ],
                        "com.apple.CoreSimulator.SimRuntime.tvOS-14-2" : [

                        ]
                      }
                    }
[+1027 ms] Command line invocation:
                        /Volumes/Extreme/Xcode.app/Contents/Developer/usr/bin/xcodebuild -list -project Runner.xcodeproj

                    Information about project "Runner":
                        Targets:
                            Runner
                            Flutter Assemble

                        Build Configurations:
                            Debug
                            Release
                            Profile

                        If no build configuration is specified and -scheme is not passed then "Release" is used.

                        Schemes:
                            Flutter Assemble
                            Runner
[   +2 ms] Building macOS application...
[   +3 ms] executing: /usr/bin/env xcrun xcodebuild -workspace
/Users/tahatesser/AndroidStudioProjects/master_flutter/macos/Runner.xcworkspace -configuration Release -scheme Runner
-derivedDataPath /Users/tahatesser/AndroidStudioProjects/master_flutter/build/macos
OBJROOT=/Users/tahatesser/AndroidStudioProjects/master_flutter/build/macos/Build/Intermediates.noindex
SYMROOT=/Users/tahatesser/AndroidStudioProjects/master_flutter/build/macos/Build/Products VERBOSE_SCRIPT_LOGGING=YES
COMPILER_INDEX_STORE_ENABLE=NO
[ +565 ms] Command line invocation:
[        ]     /Volumes/Extreme/Xcode.app/Contents/Developer/usr/bin/xcodebuild -workspace
/Users/tahatesser/AndroidStudioProjects/master_flutter/macos/Runner.xcworkspace -configuration Release -scheme Runner
-derivedDataPath /Users/tahatesser/AndroidStudioProjects/master_flutter/build/macos
OBJROOT=/Users/tahatesser/AndroidStudioProjects/master_flutter/build/macos/Build/Intermediates.noindex
SYMROOT=/Users/tahatesser/AndroidStudioProjects/master_flutter/build/macos/Build/Products VERBOSE_SCRIPT_LOGGING=YES
COMPILER_INDEX_STORE_ENABLE=NO
[   +2 ms] User defaults from command line:
[        ]     IDEDerivedDataPathOverride = /Users/tahatesser/AndroidStudioProjects/master_flutter/build/macos
[        ] Build settings from command line:
[        ]     COMPILER_INDEX_STORE_ENABLE = NO
[        ]     OBJROOT = /Users/tahatesser/AndroidStudioProjects/master_flutter/build/macos/Build/Intermediates.noindex
[        ]     SYMROOT = /Users/tahatesser/AndroidStudioProjects/master_flutter/build/macos/Build/Products
[        ]     VERBOSE_SCRIPT_LOGGING = YES
[ +436 ms] note: Using new build system
[        ] note: Building targets in parallel
[   +7 ms] note: Planning build
[ +226 ms] [
                      {
                        "simulator" : true,
                        "operatingSystemVersion" : "14.2 (18B79)",
                        "available" : true,
                        "platform" : "com.apple.platform.iphonesimulator",
                        "modelCode" : "iPhone13,2",
                        "identifier" : "B46BCEDD-9601-4CD0-8D0B-CE4EF96ACD93",
                        "architecture" : "x86_64",
                        "modelUTI" : "com.apple.iphone-12-1",
                        "modelName" : "iPhone 12",
                        "name" : "iPhone 12"
                      },
                      {
                        "simulator" : true,
                        "operatingSystemVersion" : "13.7 (17H22)",
                        "available" : true,
                        "platform" : "com.apple.platform.iphonesimulator",
                        "modelCode" : "iPhone12,8",
                        "identifier" : "F0FD09FB-9ABB-402D-9712-7AC7278ED742",
                        "architecture" : "x86_64",
                        "modelUTI" : "com.apple.iphone-se-1",
                        "modelName" : "iPhone SE (2nd generation)",
                        "name" : "iPhone SE (2nd generation)"
                      }
                    ]
[   +1 ms] [
             {
               "simulator" : true,
               "operatingSystemVersion" : "14.2 (18B79)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPhone13,2",
               "identifier" : "B46BCEDD-9601-4CD0-8D0B-CE4EF96ACD93",
               "architecture" : "x86_64",
               "modelUTI" : "com.apple.iphone-12-1",
               "modelName" : "iPhone 12",
               "name" : "iPhone 12"
             },
             {
               "simulator" : true,
               "operatingSystemVersion" : "13.7 (17H22)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPhone12,8",
               "identifier" : "F0FD09FB-9ABB-402D-9712-7AC7278ED742",
               "architecture" : "x86_64",
               "modelUTI" : "com.apple.iphone-se-1",
               "modelName" : "iPhone SE (2nd generation)",
               "name" : "iPhone SE (2nd generation)"
             }
           ]
[+1326 ms] note: Constructing build description
[ +534 ms] error: The linked and embedded framework 'App.framework' is missing one or more architectures required by
this target: arm64. (in target 'Runner' from project 'Runner')
[        ] error: The linked and embedded framework 'FlutterMacOS.framework' is missing one or more architectures
required by this target: arm64. (in target 'Runner' from project 'Runner')
[   +1 ms] ** BUILD FAILED **
[  +27 ms] Building macOS application... (completed in 3.1s)
[   +4 ms] Exception: Build process failed
[   +3 ms] "flutter run" took 6,831ms.
[  +10 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:619:7)
           <asynchronous suspension>
           #2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:1104:12)
           <asynchronous suspension>
           #3      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:959:27)
           <asynchronous suspension>
           #4      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #5      AppContext.run (package:flutter_tools/src/base/context.dart:149:12)
           <asynchronous suspension>
           #6      CommandRunner.runCommand (package:args/command_runner.dart:197:13)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand.<anonymous closure>
           (package:flutter_tools/src/runner/flutter_command_runner.dart:264:9)
           <asynchronous suspension>
           #8      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #9      AppContext.run (package:flutter_tools/src/base/context.dart:149:12)
           <asynchronous suspension>
           #10     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:220:5)
           <asynchronous suspension>
           #11     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:63:9)
           <asynchronous suspension>
           #12     run.<anonymous closure> (package:flutter_tools/runner.dart:61:12)
           <asynchronous suspension>
           #13     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #14     AppContext.run (package:flutter_tools/src/base/context.dart:149:12)
           <asynchronous suspension>
           #15     runInContext (package:flutter_tools/src/context_runner.dart:70:10)
           <asynchronous suspension>
           #16     main (package:flutter_tools/executable.dart:90:3)
           <asynchronous suspension>
           
           
[ +263 ms] ensureAnalyticsSent: 255ms
[   +2 ms] Running shutdown hooks
[        ] Shutdown hook priority 4
[   +2 ms] Shutdown hooks complete
[        ] exiting with code 1
```

</details>


<details> 
<summary>multiple flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel dev, 1.24.0-7.0.pre, on macOS 11.0.1 20B29 darwin-x64,
    locale en-GB)
    • Flutter version 1.24.0-7.0.pre at /Users/tahatesser/Code/flutter_dev
    • Framework revision a0860f6e87 (2 weeks ago), 2020-10-29 20:07:34 -0700
    • Engine revision 073263e39d
    • Dart version 2.11.0 (build 2.11.0-260.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.2)
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.2, Build version 12B45b
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.51.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (4 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D • android-arm64  • Android 10 (API 29)
    • macOS (desktop)  • macos            • darwin-x64     • macOS 11.0.1 20B29
      darwin-x64
    • Web Server (web) • web-server       • web-javascript • Flutter Tools
    • Chrome (web)     • chrome           • web-javascript • Google Chrome
      86.0.4240.198

• No issues found!
```

```bash
[✓] Flutter (Channel master, 1.24.0-8.0.pre.252, on macOS 11.0.1 20B29
    darwin-x64, locale en-GB)
    • Flutter version 1.24.0-8.0.pre.252 at
      /Users/tahatesser/Code/flutter_master
    • Framework revision bf2c9dfc05 (34 hours ago), 2020-11-14 22:38:02 -0500
    • Engine revision 3e77d854a7
    • Dart version 2.12.0 (build 2.12.0-45.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.2)
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.2, Build version 12B45b
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.51.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (4 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D • android-arm64  • Android 10 (API 29)
    • macOS (desktop)  • macos            • darwin-x64     • macOS 11.0.1 20B29
      darwin-x64
    • Web Server (web) • web-server       • web-javascript • Flutter Tools
    • Chrome (web)     • chrome           • web-javascript • Google Chrome
      86.0.4240.198

• No issues found!
```

</details>

--- a/packages/flutter_tools/bin/podhelper.rb
+++ b/packages/flutter_tools/bin/podhelper.rb
@@ -86,6 +86,9 @@ def flutter_additional_macos_build_settings(target)
     # Profile can't be derived from the CocoaPods build configuration. Use release framework (for linking only).
     configuration_engine_dir = build_configuration.type == :debug ? debug_framework_dir : release_framework_dir
     build_configuration.build_settings['FRAMEWORK_SEARCH_PATHS'] = "\"#{configuration_engine_dir}\" $(inherited)"
+
+    # ARM not yet supported https://github.com/flutter/flutter/issues/69221
+    build_configuration.build_settings['EXCLUDED_ARCHS'] = 'arm64'
   end
 end
 

--- a/packages/flutter_tools/lib/src/ios/xcodeproj.dart
+++ b/packages/flutter_tools/lib/src/ios/xcodeproj.dart
@@ -206,6 +206,10 @@ List<String> _xcodeBuildSettingsLines({
       xcodeBuildSettings.add('ARCHS=$arch');
     }
   }
+  if (useMacOSConfig) {
+    // ARM not yet supported https://github.com/flutter/flutter/issues/69221
+    xcodeBuildSettings.add('EXCLUDED_ARCHS=arm64');
+  }
 
   for (final MapEntry<String, String> config in buildInfo.toEnvironmentConfig().entries) {
     xcodeBuildSettings.add('${config.key}=${config.value}');

--- a/packages/flutter_tools/lib/src/macos/build_macos.dart
+++ b/packages/flutter_tools/lib/src/macos/build_macos.dart
@@ -111,7 +111,6 @@ Future<void> buildMacOS({
       else
         '-quiet',
       'COMPILER_INDEX_STORE_ENABLE=NO',
-      'EXCLUDED_ARCHS=arm64', // TODO(jmagman): Allow ARM https://github.com/flutter/flutter/issues/69221
     ],
     trace: true,
     stdoutErrorMatcher: verboseLogging ? null : _anyOutput,

--- a/packages/flutter_tools/test/commands.shard/hermetic/build_macos_test.dart
+++ b/packages/flutter_tools/test/commands.shard/hermetic/build_macos_test.dart
@@ -97,7 +97,6 @@ void main() {
         else
           '-quiet',
         'COMPILER_INDEX_STORE_ENABLE=NO',
-        'EXCLUDED_ARCHS=arm64',
       ],
       stdout: 'STDOUT STUFF',
       onRun: () {
@@ -265,6 +264,7 @@ void main() {
       'TREE_SHAKE_ICONS=true',
       'FLUTTER_TARGET=lib/other.dart',
       'BUNDLE_SKSL_PATH=foo/bar.sksl.json',
+      'EXCLUDED_ARCHS=arm64',
     ]));
   }, overrides: <Type, Generator>{
     FileSystem: () => fileSystem,

--- a/packages/flutter_tools/test/integration.shard/build_ios_config_only_test.dart
+++ b/packages/flutter_tools/test/integration.shard/build_ios_config_only_test.dart
@@ -41,7 +41,10 @@ void main() {
 
     // Config is updated if command succeeded.
     expect(generatedConfig, exists);
-    expect(generatedConfig.readAsStringSync(), contains('DART_OBFUSCATION=true'));
+    expect(generatedConfig.readAsStringSync(), allOf(
+      contains('DART_OBFUSCATION=true'),
+      isNot(contains('EXCLUDED_ARCHS')),
+    ));
 
     // file that only exists if app was fully built.
     final File frameworkPlist = fileSystem.file(

